packages/nextjs/app/builders/0x981ab0D817710d8FFFC5693383C00D985A3BDa38/page.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import type { NextPage } from "next";
+import { useEnsAddress, useEnsAvatar } from "wagmi";
+import { BlockieAvatar } from "~~/components/scaffold-eth";
+
+const ensName = "zapaz.eth";
+
+const Zapaz: NextPage = () => {
+  const [ensAdddress, setEnsAddress] = useState<string | null>();
+  const [ensAvatar, setEnsAvatar] = useState<string | null>();
+  const { data: fetchedEnsAddress } = useEnsAddress({ name: ensName, chainId: 1 });
+  const { data: fetchedEnsAvatar } = useEnsAvatar({ name: ensName, chainId: 1 });
+  useEffect(() => setEnsAvatar(fetchedEnsAvatar), [fetchedEnsAvatar]);
+  useEffect(() => setEnsAddress(fetchedEnsAddress), [fetchedEnsAddress]);
+
+  return (
+    <>
+      <div className="m-10 mb-20">
+        <div className="bg-base-100 border-base-300 border shadow-md shadow-secondary rounded-3xl px-6 lg:px-8 mb-6 space-y-1 py-4 overflow-x-auto">
+          <BlockieAvatar address={ensAdddress || ""} ensImage={ensAvatar} size={100} />
```

Comment:
Similar to the error handling, it's a good practice to have a loading spinner or other placeholder while the data is being fetched from `useEffect`.

For example,
```javascript
if (!ensAdddress || !ensAvatar) {
     // some loading state
}
```

Reply:
added loading-spinner